### PR TITLE
Cluster upgrade feature for bare metal

### PR DESCRIPTION
**Pull request by ptrivedi**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/executables/kubectl.go [hunk 1]
```diff
@@ -1111,6 +1113,22 @@ func (k *Kubectl) GetEksaAWSIamConfig(ctx context.Context, awsIamConfigName stri
 	return response, nil
 }
 
+func (k *Kubectl) GetEksaTinkerbellDatacenterConfig(ctx context.Context, tinkerbellDatacenterConfigName string, kubeconfigFile string, namespace string) (*v1alpha1.TinkerbellDatacenterConfig, error) {
```
Comment (chrisdoherty4) [inline pkg/executables/kubectl.go:1116]: Are we able to throw some unit tests in for these?

File: pkg/executables/kubectl.go [hunk 2]
```diff
@@ -1139,6 +1141,22 @@ func (k *Kubectl) GetEksaAWSIamConfig(ctx context.Context, awsIamConfigName stri
...
+	params := []string{"get", eksaTinkerbellDatacenterResourceType, tinkerbellDatacenterConfigName, "-o", "json", "--kubeconfig", kubeconfigFile, "--namespace", namespace}
+	stdOut, err := k.Execute(ctx, params...)
+	if err != nil {
+		return nil, fmt.Errorf("error getting eksa tinkerbell datacenter config %v", err)
```
Comment (abhinavmpandey08) [inline pkg/executables/kubectl.go:1148]: We don't prepend `error` in front of every error anymore
```suggestion
		return nil, fmt.Errorf("getting eksa TinkerbellDatacenterConfig %v", err)
```

File: pkg/executables/kubectl.go [hunk 3]
```diff
@@ -1139,6 +1141,22 @@ func (k *Kubectl) GetEksaAWSIamConfig(ctx context.Context, awsIamConfigName stri
...
+	}
+
+	response := &v1alpha1.TinkerbellDatacenterConfig{}
+	err = json.Unmarshal(stdOut.Bytes(), response)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing get eksa tinkerbell datacenter config response: %v", err)
```
Comment (abhinavmpandey08) [inline pkg/executables/kubectl.go:1154]:
```suggestion
		return nil, fmt.Errorf("parsing get eksa TinkerbellDatacenterConfig response: %v", err)
```

File: pkg/executables/kubectl.go [hunk 4]
```diff
@@ -1155,6 +1173,22 @@ func (k *Kubectl) GetEksaVSphereDatacenterConfig(ctx context.Context, vsphereDat
 	return response, nil
 }
 
+func (k *Kubectl) GetEksaTinkerbellMachineConfig(ctx context.Context, tinkerbellMachineConfigName string, kubeconfigFile string, namespace string) (*v1alpha1.TinkerbellMachineConfig, error) {
+	params := []string{"get", eksaTinkerbellMachineResourceType, tinkerbellMachineConfigName, "-o", "json", "--kubeconfig", kubeconfigFile, "--namespace", namespace}
+	stdOut, err := k.Execute(ctx, params...)
+	if err != nil {
+		return nil, fmt.Errorf("error getting eksa tinkerbell machine config %v", err)
```
Comment (abhinavmpandey08) [inline pkg/executables/kubectl.go:1180]:
```suggestion
		return nil, fmt.Errorf("getting eksa TinkerbellMachineConfig %v", err)
```

File: pkg/executables/kubectl.go [hunk 5]
```diff
@@ -1155,6 +1173,22 @@ func (k *Kubectl) GetEksaVSphereDatacenterConfig(ctx context.Context, vsphereDat
...
+	}
+
+	response := &v1alpha1.TinkerbellMachineConfig{}
+	err = json.Unmarshal(stdOut.Bytes(), response)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing get eksa tinkerbell machine config response: %v", err)
```
Comment (abhinavmpandey08) [inline pkg/executables/kubectl.go:1186]:
```suggestion
		return nil, fmt.Errorf("parsing get eksa TinkerbellMachineConfig response: %v", err)
```

**Review by abhinavmpandey08 — COMMENTED** [review]
looks good, just some minor nits

**Review by abhinavmpandey08 — APPROVED** [review]
/lgtm

**Pull request merged**